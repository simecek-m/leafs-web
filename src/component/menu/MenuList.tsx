import clsx from "clsx";
import { FC } from "react";

interface MenuListProps {
  visible: boolean;
  children?: React.ReactNode;
}

const MenuList: FC<MenuListProps> = ({ visible, children }) => {
  return (
    <div
      className={clsx(
        "absolute top-full right-0 z-10 mt-4 flex-col gap-1 rounded-3xl bg-white p-1 shadow-lg transition duration-300 dark:bg-gray-900 md:mt-3",
        {
          hidden: !visible,
        }
      )}
    >
      {children}
    </div>
  );
};

export default MenuList;
