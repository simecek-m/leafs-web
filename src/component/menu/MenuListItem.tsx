import { IconProp } from "@fortawesome/fontawesome-svg-core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { FC } from "react";

interface MenuListItemProps {
  icon: IconProp;
  children: string;
  onClick: () => void;
}

const MenuListItem: FC<MenuListItemProps> = ({ icon, children, onClick }) => {
  return (
    <button
      onClick={onClick}
      className="flex w-full cursor-pointer flex-row items-center gap-2 rounded-xl px-3 py-2 hover:bg-slate-200 dark:hover:bg-slate-800"
    >
      <FontAwesomeIcon icon={icon} />
      <span>{children}</span>
    </button>
  );
};

export default MenuListItem;
