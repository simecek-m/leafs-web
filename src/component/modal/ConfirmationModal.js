import React from "react";
import "component/modal/ConfirmationModal.sass";
import { withTranslation } from "react-i18next";
import { withModal } from "modal/withModal";

export function ConfirmationModal({ t, modal = {}, closeModal = () => {} }) {
  return modal.visible ? (
    <div className="confirmation-modal-component" onClick={() => closeModal()}>
      <div className="modal-body" onClick={e => e.stopPropagation()}>
        <h2 className="modal-title">{modal.title}</h2>
        <div className="modal-description">{modal.text}</div>
        <div className="modal-confirmation-button">
          <span
            className="confirmation-text"
            onClick={() => {
              modal.onConfirm();
              closeModal();
            }}
          >
            {t("modal.confirmation")}
          </span>
        </div>
      </div>
    </div>
  ) : null;
}

export default withModal(withTranslation()(ConfirmationModal));
