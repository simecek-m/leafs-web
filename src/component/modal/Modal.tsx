import { AnimatePresence } from "framer-motion";
import { FC, PropsWithChildren } from "react";
import { motion } from "framer-motion";

interface ModalProps extends PropsWithChildren {
  isOpen: boolean;
  onClose: () => void;
}

const Modal: FC<ModalProps> = ({ isOpen, onClose, children }) => {
  return (
    <AnimatePresence>
      {isOpen && (
        <motion.div
          initial={{ opacity: 0 }}
          animate={{ opacity: 1 }}
          exit={{ opacity: 0 }}
          className="absolute top-0 left-0 z-40 flex h-screen w-screen items-center justify-center bg-black bg-opacity-70"
          onClick={onClose}
        >
          <motion.div
            initial={{ y: -100 }}
            animate={{ y: 0 }}
            exit={{ y: -100 }}
            className="m-2 flex w-full max-w-3xl flex-col rounded-3xl bg-white p-6 dark:bg-gray-800 md:w-1/2"
            onClick={(e) => e.stopPropagation()}
          >
            {children}
          </motion.div>
        </motion.div>
      )}
    </AnimatePresence>
  );
};

export default Modal;
