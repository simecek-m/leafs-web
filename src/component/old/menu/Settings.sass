@import "colors.sass"

.settings-component
  position: fixed
  top: 0
  right: 0
  transition: 0.5s ease all

  &.expanded
    .settings-component-switch
      right: 60px

    .settings-component-sidebar
      width: 60px

  .settings-component-switch
    position: absolute
    top: 0
    right: 10px
    display: inline-block
    margin: 20px
    width: 35px
    height: 35px
    line-height: 35px
    text-align: center
    border-radius: 100%
    color: $text-primary
    padding: 10px
    cursor: pointer
    font-size: 30px
    transition: 0.5s ease all

    &:hover
      background-color: $background-secondary
      box-shadow: 5px 5px 10px $shadow

  .settings-component-sidebar
    display: inline-block
    width: 0
    height: 100vh
    background-color: $background-secondary
    transition: 0.5s ease all
    text-align: center
    box-shadow: 0 0 10px $shadow

  hr
    border-color: $text-secondary
    width: 80%
