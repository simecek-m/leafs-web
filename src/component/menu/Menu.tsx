import { IconProp } from "@fortawesome/fontawesome-svg-core";
import MenuList from "component/menu/MenuList";
import MenuListItem from "component/menu/MenuListItem";
import { FC, useCallback, useEffect, useRef, useState } from "react";

export interface IMenuListItem {
  icon: IconProp;
  text: string;
  onClick: () => void;
}

interface MenuProps {
  children: React.ReactNode;
  items: IMenuListItem[];
}

const Menu: FC<MenuProps> = ({ children, items }) => {
  const [isOpen, setOpen] = useState<boolean>(false);
  const ref = useRef<HTMLDivElement>(null);

  const handleClickOutside = useCallback((e: MouseEvent) => {
    if (ref.current && !ref.current.contains(e.target as HTMLElement)) {
      setOpen(false);
    }
  }, []);

  // encapsulate to useClickOutside hook
  useEffect(() => {
    isOpen
      ? document.addEventListener("click", handleClickOutside)
      : document.removeEventListener("click", handleClickOutside);
    return () =>
      document.removeEventListener("click", handleClickOutside, true);
  }, [isOpen, handleClickOutside]);

  return (
    <div className="relative" ref={ref}>
      <button onClick={() => setOpen(!isOpen)}>{children}</button>
      <MenuList visible={isOpen}>
        {items.map((item, index) => (
          <MenuListItem
            key={index}
            icon={item.icon}
            onClick={() => {
              item.onClick();
              setOpen(false);
            }}
          >
            {item.text}
          </MenuListItem>
        ))}
      </MenuList>
    </div>
  );
};

export default Menu;
