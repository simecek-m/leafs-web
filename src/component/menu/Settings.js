import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCog } from "@fortawesome/free-solid-svg-icons";
import { faTimes } from "@fortawesome/free-solid-svg-icons";
import { SELECT } from "modal/types";
import csFlag from "assets/image/flag/cs.png";
import enFlag from "assets/image/flag/en.png";
import i18n from "i18n";
import { withModal } from "modal/withModal";
import { withTranslation } from "react-i18next";
import SettingsItem from "component/menu/SettingsItem";
import {
  faHome,
  faUser,
  faFlagUsa,
  faInfo
} from "@fortawesome/free-solid-svg-icons";
import { redirect } from "helper/router";
import "component/menu/Settings.sass";

export class Settings extends React.Component {
  state = {
    expanded: false
  };

  switchExpanded() {
    const expanded = this.state.expanded;
    this.setState({
      expanded: !expanded
    });
  }

  openChangeLanguageModal() {
    const { t, openModal } = this.props;
    openModal({
      type: SELECT,
      title: t("settings.localization.modal.title"),
      text: t("settings.localization.modal.text"),
      options: [
        {
          image: enFlag,
          text: t("settings.localization.modal.language.english"),
          onClick: () => i18n.changeLanguage("en")
        },
        {
          image: csFlag,
          text: t("settings.localization.modal.language.czech"),
          onClick: () => i18n.changeLanguage("cs")
        }
      ]
    });
  }

  render() {
    const settingsIcon = this.state.expanded ? faTimes : faCog;
    const divider = this.props.children ? <hr /> : null;
    return (
      <div
        className={`settings-component ${
          this.state.expanded ? "expanded" : ""
        }`}
      >
        <FontAwesomeIcon
          className="settings-component-button"
          icon={settingsIcon}
          onClick={() => this.switchExpanded()}
        />
        <div className="settings-component-sidebar">
          <div className="common-actions">
            <SettingsItem icon={faHome} onClick={() => redirect("/")} />
            <SettingsItem icon={faUser} onClick={() => redirect("/profile")} />
            <SettingsItem
              icon={faFlagUsa}
              onClick={() => this.openChangeLanguageModal()}
            />
            <SettingsItem icon={faInfo} onClick={() => redirect("/about")} />
          </div>
          {divider}
          <div className="component-actions">{this.props.children}</div>
        </div>
      </div>
    );
  }
}

export default withModal(withTranslation()(Settings));
