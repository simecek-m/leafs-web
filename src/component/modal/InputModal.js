import React from "react";
import { withTranslation } from "react-i18next";
import { withModal } from "modal/withModal";
import "component/modal/InputModal.sass";

export function InputModal({ t, modal = {}, closeModal }) {
  const modalInput = React.createRef();
  return modal.visible ? (
    <div className="input-modal-component" onClick={() => closeModal()}>
      <div className="modal-body" onClick={e => e.stopPropagation()}>
        <h2 className="modal-title">{modal.title}</h2>
        <div className="modal-description">{modal.text}</div>
        <input ref={modalInput} className="modal-input" autoFocus />
        <div className="modal-confirmation-button">
          <span
            className="confirmation-text"
            onClick={() => {
              modal.onConfirm(modalInput.current.value);
              closeModal();
            }}
          >
            {t("modal.confirmation")}
          </span>
        </div>
      </div>
    </div>
  ) : null;
}

export default withModal(withTranslation()(InputModal));
