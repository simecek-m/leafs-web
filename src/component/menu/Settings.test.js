import React from "react";
import { Settings } from "component/menu/Settings";
import { shallow } from "enzyme";
import { SELECT } from "modal/types";
import csFlag from "assets/image/flag/cs.png";
import enFlag from "assets/image/flag/en.png";
import * as helper from "helper/router";

jest.mock("react-i18next", () => ({
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: key => key };
    return Component;
  }
}));

const redirectSpy = jest.spyOn(helper, "redirect");

afterEach(() => {
  redirectSpy.mockReset();
});

describe("Settings component", () => {
  test("should render default Settings", () => {
    const wrapper = shallow(<Settings />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should render Settings after expand", () => {
    const wrapper = shallow(<Settings />);
    const icon = wrapper.find(".settings-component-switch");
    icon.simulate("click");
    expect(wrapper).toMatchSnapshot();
  });

  test("should set default expanded state to false", () => {
    const wrapper = shallow(<Settings />);
    const state = wrapper.state();
    expect(state.expanded).toBeFalsy();
  });

  test("should switch between expanded states (true / false)", async () => {
    const wrapper = await shallow(<Settings />);
    let state = wrapper.state();
    expect(state.expanded).toBeFalsy();
    const instance = wrapper.instance();
    await instance.switchExpanded();
    state = wrapper.state();
    expect(state.expanded).toBeTruthy();
    await instance.switchExpanded();
    state = wrapper.state();
    expect(state.expanded).toBeFalsy();
  });

  test("should open modal for change language", async () => {
    const openModalMock = jest.fn();
    const wrapper = await shallow(<Settings openModal={openModalMock} />);
    const instance = wrapper.instance();
    await instance.openChangeLanguageModal();
    expect(openModalMock).toHaveBeenCalledTimes(1);
    expect(openModalMock).toHaveBeenCalledWith({
      type: SELECT,
      title: "settings.localization.modal.title",
      text: "settings.localization.modal.text",
      options: [
        {
          image: enFlag,
          text: "settings.localization.modal.language.english",
          onClick: expect.any(Function)
        },
        {
          image: csFlag,
          text: "settings.localization.modal.language.czech",
          onClick: expect.any(Function)
        }
      ]
    });
    openModalMock.mockRestore();
  });

  test("should redirect to /", async () => {
    const wrapper = await shallow(<Settings />);
    const elem = wrapper.find("#settings-item-home");
    await elem.simulate("click");
    expect(redirectSpy).toHaveBeenCalledTimes(1);
    expect(redirectSpy).toHaveBeenCalledWith("/");
  });

  test("should redirect to /profile", async () => {
    const wrapper = await shallow(<Settings />);
    const elem = wrapper.find("#settings-item-profile");
    await elem.simulate("click");
    expect(redirectSpy).toHaveBeenCalledTimes(1);
    expect(redirectSpy).toHaveBeenCalledWith("/profile");
  });

  test("should redirect to /profile", async () => {
    const openModalMock = jest.fn();
    const wrapper = await shallow(<Settings openModal={openModalMock} />);
    const elem = wrapper.find("#settings-item-language");
    await elem.simulate("click");
    expect(openModalMock).toHaveBeenCalledTimes(1);
    openModalMock.mockRestore();
  });

  test("should redirect to /about", async () => {
    const wrapper = await shallow(<Settings />);
    const elem = wrapper.find("#settings-item-about");
    await elem.simulate("click");
    expect(redirectSpy).toHaveBeenCalledTimes(1);
    expect(redirectSpy).toHaveBeenCalledWith("/about");
  });
});
