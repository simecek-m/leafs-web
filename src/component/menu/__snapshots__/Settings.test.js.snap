// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component should render Settings after expand 1`] = `
<div
  className="settings-component expanded"
>
  <FontAwesomeIcon
    className="settings-component-switch"
    icon={
      Object {
        "icon": Array [
          352,
          512,
          Array [],
          "f00d",
          "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
        ],
        "iconName": "times",
        "prefix": "fas",
      }
    }
    onClick={[Function]}
  />
  <div
    className="settings-component-sidebar"
  >
    <div
      className="common-actions"
    >
      <SettingsItem
        icon={
          Object {
            "icon": Array [
              576,
              512,
              Array [],
              "f015",
              "M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z",
            ],
            "iconName": "home",
            "prefix": "fas",
          }
        }
        id="settings-item-home"
        onClick={[Function]}
      />
      <SettingsItem
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f007",
              "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
            ],
            "iconName": "user",
            "prefix": "fas",
          }
        }
        id="settings-item-profile"
        onClick={[Function]}
      />
      <SettingsItem
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f74d",
              "M32 0C14.3 0 0 14.3 0 32v464c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V32C64 14.3 49.7 0 32 0zm267.9 303.6c-57.2-15.1-111.7-28.8-203.9 11.1V384c185.7-92.2 221.7 53.3 397.5-23.1 11.4-5 18.5-16.5 18.5-28.8v-36c-43.6 17.3-80.2 24.1-112.1 24.1-37.4-.1-68.9-8.4-100-16.6zm0-96c-57.2-15.1-111.7-28.8-203.9 11.1v61.5c94.8-37.6 154.6-22.7 212.1-7.6 57.2 15.1 111.7 28.8 203.9-11.1V200c-43.6 17.3-80.2 24.1-112.1 24.1-37.4 0-68.9-8.3-100-16.5zm9.5-125.9c51.8 15.6 97.4 29 202.6-20.1V30.8c0-25.1-26.8-38.1-49.4-26.6C291.3 91.5 305.4-62.2 96 32.4v151.9c94.8-37.5 154.6-22.7 212.1-7.6 57.2 15 111.7 28.7 203.9-11.1V96.7c-53.6 23.5-93.3 31.4-126.1 31.4s-59-7.8-85.7-15.9c-4-1.2-8.1-2.4-12.1-3.5V75.5c7.2 2 14.3 4.1 21.3 6.2zM160 128.1c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16-7.2 16-16 16zm0-55.8c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16c0 8.8-7.2 16-16 16zm64 47.9c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16c0 8.8-7.2 16-16 16zm0-55.9c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16c0 8.8-7.2 16-16 16z",
            ],
            "iconName": "flag-usa",
            "prefix": "fas",
          }
        }
        id="settings-item-language"
        onClick={[Function]}
      />
      <SettingsItem
        icon={
          Object {
            "icon": Array [
              192,
              512,
              Array [],
              "f129",
              "M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z",
            ],
            "iconName": "info",
            "prefix": "fas",
          }
        }
        id="settings-item-about"
        onClick={[Function]}
      />
    </div>
    <div
      className="component-actions"
    />
  </div>
</div>
`;

exports[`Settings component should render default Settings 1`] = `
<div
  className="settings-component "
>
  <FontAwesomeIcon
    className="settings-component-switch"
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f013",
          "M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z",
        ],
        "iconName": "cog",
        "prefix": "fas",
      }
    }
    onClick={[Function]}
  />
  <div
    className="settings-component-sidebar"
  >
    <div
      className="common-actions"
    >
      <SettingsItem
        icon={
          Object {
            "icon": Array [
              576,
              512,
              Array [],
              "f015",
              "M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z",
            ],
            "iconName": "home",
            "prefix": "fas",
          }
        }
        id="settings-item-home"
        onClick={[Function]}
      />
      <SettingsItem
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f007",
              "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
            ],
            "iconName": "user",
            "prefix": "fas",
          }
        }
        id="settings-item-profile"
        onClick={[Function]}
      />
      <SettingsItem
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f74d",
              "M32 0C14.3 0 0 14.3 0 32v464c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V32C64 14.3 49.7 0 32 0zm267.9 303.6c-57.2-15.1-111.7-28.8-203.9 11.1V384c185.7-92.2 221.7 53.3 397.5-23.1 11.4-5 18.5-16.5 18.5-28.8v-36c-43.6 17.3-80.2 24.1-112.1 24.1-37.4-.1-68.9-8.4-100-16.6zm0-96c-57.2-15.1-111.7-28.8-203.9 11.1v61.5c94.8-37.6 154.6-22.7 212.1-7.6 57.2 15.1 111.7 28.8 203.9-11.1V200c-43.6 17.3-80.2 24.1-112.1 24.1-37.4 0-68.9-8.3-100-16.5zm9.5-125.9c51.8 15.6 97.4 29 202.6-20.1V30.8c0-25.1-26.8-38.1-49.4-26.6C291.3 91.5 305.4-62.2 96 32.4v151.9c94.8-37.5 154.6-22.7 212.1-7.6 57.2 15 111.7 28.7 203.9-11.1V96.7c-53.6 23.5-93.3 31.4-126.1 31.4s-59-7.8-85.7-15.9c-4-1.2-8.1-2.4-12.1-3.5V75.5c7.2 2 14.3 4.1 21.3 6.2zM160 128.1c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16-7.2 16-16 16zm0-55.8c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16c0 8.8-7.2 16-16 16zm64 47.9c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16c0 8.8-7.2 16-16 16zm0-55.9c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16c0 8.8-7.2 16-16 16z",
            ],
            "iconName": "flag-usa",
            "prefix": "fas",
          }
        }
        id="settings-item-language"
        onClick={[Function]}
      />
      <SettingsItem
        icon={
          Object {
            "icon": Array [
              192,
              512,
              Array [],
              "f129",
              "M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z",
            ],
            "iconName": "info",
            "prefix": "fas",
          }
        }
        id="settings-item-about"
        onClick={[Function]}
      />
    </div>
    <div
      className="component-actions"
    />
  </div>
</div>
`;
